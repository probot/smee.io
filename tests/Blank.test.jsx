import React from 'react'
import Blank from '../src/components/Blank'
import { shallow } from 'enzyme'

describe('<Blank />', () => {
  beforeEach(() => {
    Object.defineProperties(window, {
      location: {
        value: {
          href: 'https:/smee.io/CHANNEL'
        }
      }
    })
  })

  describe('render', () => {
    it('renders the blank page', () => {
      expect(<Blank />).toMatchSnapshot()
    })

    it('selects the input text when focused', () => {
      const spy = jest.fn()
      const wrapper = shallow(<Blank />)

      wrapper.find('input').simulate('focus', { target: { select: spy } })
      expect(spy).toHaveBeenCalled()
    })
  })
})
