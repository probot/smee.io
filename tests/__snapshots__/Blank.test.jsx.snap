// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Blank /> render renders the blank page 1`] = `<Blank />`;
